CA1721 (Property names should not match get methods) fires on every type that has a property named Type

#### Analyzer package

Microsoft.CodeQuality.Analyzers

#### Analyzer

PropertyNamesShouldNotMatchGetMethodsAnalyzer

#### Repro steps

```
class T { }
class C
{
    public T Type { get; }
}
```

#### Expected behavior
No CA1721. FxCop does not fire for this case.

#### Actual behavior
`warning CA1721: The property name 'Type' is confusing given the existence of method 'GetType'. Rename or remove one of these members.`

We should ignore `Object.GetType` in our analysis. Otherwise, this rule is just noise.

--- a/src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/PropertyNamesShouldNotMatchGetMethods.cs
+++ b/src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/PropertyNamesShouldNotMatchGetMethods.cs
@@ -84,7 +84,7 @@ namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines
                 {
                     // Ignore Object.GetType, as it's commonly seen and Type is a commonly-used property name.
                     if (member.ContainingType.SpecialType == SpecialType.System_Object &&
-                        member.Name == "GetType")
+                        member.Name == nameof(GetType))
                     {
                         continue;
                     }

--- a/src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/PropertyNamesShouldNotMatchGetMethods.cs
+++ b/src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/PropertyNamesShouldNotMatchGetMethods.cs
@@ -82,6 +82,13 @@ namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines
                 var exposedMembers = type.GetMembers(identifier).Where(member => ExposedAccessibilities.Contains(member.DeclaredAccessibility));
                 foreach (var member in exposedMembers)
                 {
+                    // Ignore Object.GetType, as it's commonly seen and Type is a commonly-used property name.
+                    if (member.ContainingType.SpecialType == SpecialType.System_Object &&
+                        member.Name == "GetType")
+                    {
+                        continue;
+                    }
+
                     // If the declared type is a property, was a matching method found?
                     if (symbol.Kind == SymbolKind.Property && member.Kind == SymbolKind.Method)
                     {

--- a/src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/PropertyNamesShouldNotMatchGetMethodsTests.cs
+++ b/src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/PropertyNamesShouldNotMatchGetMethodsTests.cs
@@ -278,6 +278,24 @@ End Class",
             GetCA1721BasicResultAt(line: 13, column: 21, identifierName: "Date", otherIdentifierName: "GetDate"));
         }
 
+        [Fact, WorkItem(1374, "https://github.com/dotnet/roslyn-analyzers/issues/1374")]
+        public void CA1721_TypePropertyNoDiagnostic()
+        {
+            VerifyCSharp(@"
+class T { }
+class C
+{
+    public T Type { get; }
+}");
+
+            VerifyBasic(@"
+Class T
+End Class
+Class C
+    Public Property Type As T
+End Class");
+        }
+
         #region Helpers
 
         private static DiagnosticResult GetCA1721CSharpResultAt(int line, int column, string identifierName, string otherIdentifierName)

